### ISSUE TITLE
The requested module 'elliptic' is a CommonJS module

### ISSUE THREAD

**Issue by 0xYami** [FuelLabs/fuels-ts#841]
### What version of fuels-ts are you using?

^0.35.0

### Steps to Reproduce

Here's a simple NextJS [Stackblitz example](https://stackblitz.com/edit/nextjs-zwgubp?file=pages/index.js) with only `fuels` installed

1 - Open the link
2 - Wait for the installation to finish
3 - The app gets a runtime error

For the Vitest example, here's the branch with a simple test for getting the current block number https://github.com/0xYami/fuels-react/tree/feat/unit-tests

1 - Clone the repository
2 - Checkout the `feat/unit-tests` branch
3 - Install dependencies - `pnpm i`
4 - Run local testnet - `fuel-core run --db-type in-memory`
5 - Run tests - `pnpm test`
6 - The same error as above occurs

### Expected Behavior

To be able to use in Vitest and NextJS

### Actual Behavior

I'm running into an issue while writing unit tests using [Vitest](https://vitest.dev) for the [React library](https://fuels-react.com) I'm currently working on. I'm not sure exactly but the error seems related to some CommonJS dependencies not correctly bundled (in this case the `elliptic` dependency) to work properly in ESM.

I tried a few things and noticed that a fresh NextJS project with only `fuels` installed (like in the [reproduction](https://stackblitz.com/edit/nextjs-zwgubp?file=pages/index.js)) crashes when using the `Signer` class which imports the `elliptic` library.

And while trying to debug a bit more I also noticed that having the `fuels-ts` repository locally and linking the `fuels` dependency in my React library directly to it make things work.

Here's the error I'm getting in both Vitest and NextJS.

```
SyntaxError: Named export 'ec' not found. The requested module 'elliptic' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'elliptic';
const {concat: n,hexlify: c,arrayify: s}from"@ethersproject/bytes";import{Address: l}from"@fuel-ts/address";import{hash: b}from"@fuel-ts/hasher";import{randomBytes: u}from"@fuel-ts/keystore";import{toBytes: d}from"@fuel-ts/math";import{ec: p}from"elliptic";function o(){return new p("secp256k1")}var a=class{constructor(e){typeof e=="string"&&e.match(/^[0-9a-f]*$/i)&&e.length===64&&(e=`0x${e}`);let r=s(e),t=o().keyFromPrivate(r,"hex");this.compressedPublicKey=c(t.getPublic(!0,"array")),this.publicKey=c(t.getPublic(!1,"array").slice(1)),this.privateKey=c(r),this.address=l.fromPublicKey(this.publicKey)}sign(e){let t=o().keyFromPrivate(s(this.privateKey),"hex").sign(s(e),{canonical:!0}),y=d(t.r,32),i=d(t.s,32);return i[0]|=(t.recoveryParam||0)<<7,c(n([y,i]))}addPoint(e){let r=o().keyFromPublic(s(this.compressedPublicKey)),t=o().keyFromPublic(s(e)),y=r.getPublic().add(t.getPublic());return c(y.encode("array",!0))}static recoverPublicKey(e,r){let t=s(r),y=t.slice(0,32),i=t.slice(32,64),m=(i[0]&128)>>7;return i[0]&=127,o().recoverPubKey(s(e),{r:y,s:i},m).encode("array",!1).slice(1)}static recoverAddress(e,r){return l.fromPublicKey(a.recoverPublicKey(e,r))}static generatePrivateKey(e){return e?b(n([u(32),s(e)])):u(32)}static extendPublicKey(e){let r=o().keyFromPublic(s(e));return c(r.getPublic(!1,"array").slice(1))}},k=a;export{k: Signer,o: getCurve} = pkg;
```

**Comment by 0xYami** [comment]
@arboleya It looks like the wallet only uses `CSR` with Vite, I don't have this issue on my Vite [playground](https://stackblitz.com/fork/github/0xYami/fuels-react/tree/main/playgrounds/vite?file=src/main.tsx) so it might be related to Vitest/Next doing some stuff on Node env

**Comment by Dhaiwat10** [comment]
I tried using the SDK in a fresh Next.js app, and I am getting the same error. Link for repro: https://github.com/Dhaiwat10/fuels-nextjs-bug-repro

**Comment by diyahir** [comment]
I am experiencing the same issue

**Comment by LuizAsFight** [comment]
> @luizstacio @LuizAsFight Did you experience anything similar to this on the wallet?

nop

**Comment by Dierkalalala** [comment]
Im having this issue as well

**Comment by Dhaiwat10** [comment]
Keeping this issue open until we publish a new release and verify that this has been fixed.

**Comment by arboleya** [comment]
Released in `0.38.1`:
 - https://github.com/FuelLabs/fuels-ts/pull/883

**Comment by Dhaiwat10** [comment]
Thanks @arboleya, the issue seems to be fixed. 🎊
